### PR TITLE
Use ExternalID in NodeStageVolume RPC

### PR DESCRIPTION
**Pull request by angrycub**
This resolves an issue with the gcp-compute-persistent-disk-csi-driver where a volume will fail to mount with the following error:

```
2020-04-20T13:25:29Z  Setup Failure  failed to setup alloc: pre-run hook "csi_hook" failed: rpc error: code = InvalidArgument desc = NodeStageVolume Volume ID is invalid: failed to get id components. Expected projects/{project}/zones/{zone}/disks/{name}. Got: mysql
```

### CODE REVIEW

**Comment by angrycub** [comment]
In my test, the volume was defined as follows:
```
type = "csi"
id = "mysql"
name = "mysql"
#selfLink from the json
external_id = "projects/my_project/zones/us-east1-b/disks/cv-disk-1"
access_mode = "single-node-writer"
attachment_mode = "file-system"
plugin_id = "gcepd"
```

File: client/pluginmanager/csimanager/volume.go [hunk 1]
```diff
@@ -166,7 +166,7 @@ func (v *volumeManager) stageVolume(ctx context.Context, vol *structs.CSIVolume,
 	// CSI NodeStageVolume errors for timeout, codes.Unavailable and
 	// codes.ResourceExhausted are retried; all other errors are fatal.
 	return v.plugin.NodeStageVolume(ctx,
-		vol.ID,
+		vol.ExternalID,
```
Comment (tgross) [inline client/pluginmanager/csimanager/volume.go:169]: There's a getter method `vol.RemoteID()` for this. Let's use that here for consistency with the `v.plugin.NodePublishVolume` call.

**Comment by angrycub** [comment]
Manually tested AWS EBS using the original fix version (using vol.ExternalID) and it does not regress it. Perhaps the AWS EBS volume driver doesn't actually cause staging to happen?

**Comment by tgross** [comment]
> Perhaps the AWS EBS volume driver doesn't actually cause staging to happen?

Yeah, entirely possible that it supports the RPC but only actually does the staging work under specific circumstances.

**Comment by tgross** [comment]
> Yeah, entirely possible that it supports the RPC but only actually does the staging work under specific circumstances.

Was just digging through the code and it looks like this might also help explain the NVME issue you were reporting in chat: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/04f0e8a8c70a8b4ba0fa76b24f2aef7bc26a18cc/pkg/driver/node.go#L479 (assuming the Docker folks fixed the problem with the actual mounting step).

**Pull request merged**